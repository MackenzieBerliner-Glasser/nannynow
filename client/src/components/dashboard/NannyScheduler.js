import React, { PureComponent } from 'react';
import 'react-datepicker/dist/react-datepicker-cssmodules.css';
import PropTypes from 'prop-types';
import Range from './Range';
import styles from './NannyScheduler.css';
import globalStyles from '../../main.css';
import AvailabilityDisplay from './AvailabilityDisplay';

export default class NannyScheduler extends PureComponent {
  static propTypes = {
    updateAvailability: PropTypes.func.isRequired
  };

  state = {
    start: null,
    end: null
  };

  handleStartChange = date => {
    this.setState({ start: date });
  };

  handleEndChange = date => {
    this.setState({ end: date });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { start, end } = this.state;
    const { _id } = this.props.user;
    {start && end && this.props.updateAvailability(start.toISOString(), end.toISOString(), _id);}
  };

  render() {
    const { start, end } = this.state;
    return (
      <div>
        <form onSubmit={this.handleSubmit} id={globalStyles.form} className={styles.schedulerForm}>
          <Range
            onStartChange={this.handleStartChange}
            onEndChange={this.handleEndChange}
            start={start}
            end={end}
            id={styles.range}
          />
          <button id={globalStyles.button} className={styles.addAvailButton}>Submit Availability</button>
        </form>
        <AvailabilityDisplay />
      </div>
    );
  }
}
