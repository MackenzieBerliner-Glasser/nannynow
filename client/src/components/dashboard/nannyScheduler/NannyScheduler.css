.schedulerForm {
  padding: 15px;
  width: 50%;
  margin: 15px;
  display: flex;
  flex-direction: column;

  .addAvailButton {
    height: 35px;
    width: 160px;
    align-self: center;
    font-size: 15px;
    background-color: #f96a54;
    &:hover {
      background-color: #7fbbe0;
      cursor: pointer;
    }
  }
}

@media (max-width: 700px) {
  .schedulerForm {
    width: 100%;
  }
}
