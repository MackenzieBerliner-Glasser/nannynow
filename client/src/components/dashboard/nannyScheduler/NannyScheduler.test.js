import React from 'react';
import { shallow } from 'enzyme';
import NannyScheduler from './NannyScheduler';

jest.mock('../../helpers/hand_logo.png');

describe('NannyScheduler component', () => {
  it('renders a component', () => {
    const updateAvailability = jest.fn();
    const wrapper = shallow(<NannyScheduler updateAvailability={updateAvailability} />);
    expect(wrapper).toMatchSnapshot();
  });
});
