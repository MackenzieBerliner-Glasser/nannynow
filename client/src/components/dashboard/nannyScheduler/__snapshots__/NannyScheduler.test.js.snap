// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NannyScheduler component renders a component 1`] = `
<Fragment>
  <form
    className="schedulerForm"
    id="form"
    onSubmit={[Function]}
  >
    <Range
      end={null}
      id="range"
      onEndChange={[Function]}
      onStartChange={[Function]}
      start={null}
    />
    <button
      className="addAvailButton"
      id="button"
    >
      Submit Availability
    </button>
    <Connect(AvailabilityDisplay) />
  </form>
</Fragment>
`;
